### PR TITLE
Update the Kiva GraphicsContext documentation

### PR DESCRIPTION
**Pull request by jwiggins**
Most, but not all of #618. This nearly covers all the methods of GraphicsContext.

### CODE REVIEW

File: docs/source/kiva/quickref.rst [hunk 1]
```diff
@@ -75,7 +75,11 @@ This is an internal representation of fonts for the ``kiva.agg`` backend. Use
 
 CompiledPath
 ~~~~~~~~~~~~
-Interface is the same as the `Path functions`_ in Graphics Context.
+A path is a colection of geometry that can be draw in a graphics context with
```
Comment (corranwebster) [inline docs/source/kiva/quickref.rst:78]:
```suggestion
A path is a collection of geometry that can be drawn in a graphics context with
```

File: docs/source/kiva/quickref.rst [hunk 2]
```diff
@@ -164,19 +168,32 @@ __init__(ary_or_size, pix_format="bgra32", interpolation="nearest", base_pixel_s
 State functions
 ~~~~~~~~~~~~~~~
 * ``save_state()``
```
Comment (corranwebster) [inline docs/source/kiva/quickref.rst:170]: The bullet points probably aren't needed here and below.

File: docs/source/kiva/quickref.rst [hunk 3]
```diff
@@ -315,7 +423,7 @@ Misc functions
 * ``begin_page()``
 * ``end_page()``
 * ``clear_rect(rect)``
-    Clears a rect. Not available in PDF context
+    Clears a rect. Not available in the PDF backend,
```
Comment (corranwebster) [inline docs/source/kiva/quickref.rst:426]: Possibly the postscript backend as well?

**Review by corranwebster — APPROVED** [review]

**Comment by jwiggins** [comment]
@corranwebster Thanks for the review. I actually made a major change while you were in the process of reviewing. This documentation now specifically references the `AbstractGraphicsContext` docstrings via Sphinx autodoc and the `automethod` directive. I like it _much better_ this way since it keeps the documentation close-ish to the code.

File: docs/source/kiva/quickref.rst [hunk 4]
```diff
@@ -5,6 +5,158 @@ This document is a summary of the classes and functions available in
...
+.. automethod:: kiva.abstract_graphics_context.AbstractGraphicsContext.fill_path
+.. automethod:: kiva.abstract_graphics_context.AbstractGraphicsContext.eof_fill_path
+.. automethod:: kiva.abstract_graphics_context.AbstractGraphicsContext.stroke_path
+.. automethod:: kiva.abstract_graphics_context.AbstractGraphicsContext.draw_rect
+.. automethod:: kiva.abstract_graphics_context.AbstractGraphicsContext.draw_image
+
+Enhanced drawing functions
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. note::
+   These methods are not available from every backend, so you should text for
```
Comment (jwiggins) [inline docs/source/kiva/quickref.rst:121]: Typo: text -> test

File: docs/source/kiva/quickref.rst [hunk 5]
```diff
@@ -136,206 +283,3 @@ path_cmd:
...
-__init__(ary_or_size, pix_format="bgra32", interpolation="nearest", base_pixel_scale=1.0):
-    ``ary_or_size`` can be either a numpy array or a tuple of the form
-    (width, height). If it is an array, it will be used as the backing store
-    for the pixels. **Its shape must be compatible with ``pix_format``**
-
-    ``pix_format`` determines the pixel format and is a string which can be any
-    of the following: "gray8", "rgb555", "rgb565", "rgb24", "bgr24", "rgba32",
-    "argb32", "abgr32", "bgra32".
-
-    ``interpolation`` determines the interpolation used by scaled image drawing
```
Comment (rahulporuri) [inline docs/source/kiva/quickref.rst:155]: ~not sure why we removed this from the docstring.~ looks like this isnt available consistently across the various supported backends.

**Review by rahulporuri — APPROVED** [review]
LGTM. Thanks for the docs refactor!

**Comment by jwiggins** [comment]
Thanks for the feedback. Should we consider #618 closed now? @corranwebster?

**Pull request merged**